app/src/main/java/org/thoughtcrime/securesms/database/MmsDatabase.kt

```diff
@@ -337,230 +296,176 @@ class MmsDatabase(context: Context, databaseHelper: SQLCipherOpenHelper) : Messa
-                    subscriptionId,
-                    expiresIn,
-                    expireStartedAt,
-                    distributionType,
-                    quote,
-                    contacts,
-                    previews,
-                    networkFailures!!,
-                    mismatches!!
-                )
-                return if (MmsSmsColumns.Types.isSecureType(outboxType)) {
-                    OutgoingSecureMediaMessage(message)
-                } else message
-            }
-            throw NoSuchMessageException("No record found for id: $messageId")
-        } finally {
-            cursor?.close()
+            } ?: emptyList()
+            val message = OutgoingMediaMessage(
+                recipient,
+                body,
+                attachments,
+                timestamp,
+                subscriptionId,
+                expiresIn,
+                expireStartedAt,
+                distributionType,
+                quote,
+                contacts,
+                previews,
+                networkFailures,
+                mismatches
+            )
+            return if (MmsSmsColumns.Types.isSecureType(outboxType)) {
+                OutgoingSecureMediaMessage(message)
+            } else message
         }
     }
 
     private fun getSharedContacts(
         cursor: Cursor,
         attachments: List<DatabaseAttachment>
-    ): List<Contact> {
-        val serializedContacts = cursor.getString(cursor.getColumnIndexOrThrow(SHARED_CONTACTS))
-        if (serializedContacts.isNullOrEmpty()) {
-            return emptyList()
-        }
-        val attachmentIdMap: MutableMap<AttachmentId?, DatabaseAttachment> = HashMap()
-        for (attachment in attachments) {
-            attachmentIdMap[attachment.attachmentId] = attachment
-        }
-        try {
-            val contacts: MutableList<Contact> = LinkedList()
-            val jsonContacts = JSONArray(serializedContacts)
-            for (i in 0 until jsonContacts.length()) {
-                val contact = Contact.deserialize(jsonContacts.getJSONObject(i).toString())
-                if (contact.avatar != null && contact.avatar!!.attachmentId != null) {
-                    val attachment = attachmentIdMap[contact.avatar!!.attachmentId]
-                    val updatedAvatar = Contact.Avatar(
-                        contact.avatar!!.attachmentId,
-                        attachment,
-                        contact.avatar!!.isProfile
-                    )
-                    contacts.add(Contact(contact, updatedAvatar))
-                } else {
-                    contacts.add(contact)
-                }
-            }
-            return contacts
-        } catch (e: JSONException) {
-            Log.w(TAG, "Failed to parse shared contacts.", e)
-        } catch (e: IOException) {
-            Log.w(TAG, "Failed to parse shared contacts.", e)
-        }
-        return emptyList()
+    ): List<Contact> = try {
```

Comment:
This is... tricky. Would love to test that this method still works just for peace of mind.